### ISSUE TITLE
wandb: ERROR Error while calling W&B API: Error 1406: Data too long for column 'group_name' at row 1 (<Response [500]>)

### ISSUE THREAD

**Issue by nivha** [wandb/client#1519]
It seems like there's a limit on the group name length. Length 152 works, Length 162 gets the above error message (Didn't try to serach the exact number where it fails)

If it's possible to extend the limit that would be great. Otherwise, would be nice to have a more informative error message.

Thanks.

**Comment by vanpelt** [comment]
The group_name has a maximum length of 128 characters (I think this could be more bytes with unicode).  We'll look into better error messages and the possibility to increase or do something smarter in the client library.

**Comment by nivha** [comment]
The reason I used long group name to begin with, is because I wanted to plot train-loss of different epochs on the same plot (say I have about 10 epochs). The only way I found to do it was to ```wandb.init(group='experiment-name', name='epoch#', reinit=True)``` every epoch, and since group-name holds the name for the whole experiment I wanted it to have an indicative name with all relevant parameters, thus becoming incredibly large.

If there's a better way to do it I'll be glad to know, couldn't find anything relevant in your documentation. thanks

**Comment by vanpelt** [comment]
Hey @nivha you can hash the parameters using something like `md5`.  The group name doesn't need to be human readable because you can search or filter by the actual parameters in the UI if you're logging them to config.  See this stack overflow for different ways to hash the parameters: https://stackoverflow.com/questions/5884066/hashing-a-dictionary

**Comment by nivha** [comment]
I think the poblem is mostly UI related, if you want I can open a new issue (btw if I wanted to use a non-human readable group name I could just use your ```wandb.util.generate_id()``` function, no need for fancy hashing).

Anyways, if I want to track the progress of train-loss of different epochs plotted on the same graph (say train-loss per batch/iteration) the only way I found was to reinit wandb under same group.
In this sense each epoch is a different "run" and the UI groups them under the same group name, that way I can track the progress of several experiments simultaneously.

It is indeed a bad solution since all epochs should be just one "run" and not separate ones. But I couldn't find a better solution and there's nothing in your documentation to suggest how to do it (I'm interested in tracking my experiments while they are running, and not necessarily in filtering post-running analysis).

**Comment by vanpelt** [comment]
Hey @nivha you should be able to just use wandb.log in a training loop:

```
wandb.init(project="metrics")

for epoch in range(10):
   for batch in range(10):
      wandb.log({"batch": batch + epoch * batch, "epoch": epoch, "loss": loss})
```

Then you can view the loss plotted against batch or epoch by using the "x-axis" option in the UI.

**Comment by nivha** [comment]
I'm sorry but I just can't figure out how to do it.
Here's my code like you suggested:

```
import wandb
import numpy as np
wandb.init(project='metrics')
for epoch in range(10):
   a = np.random.rand(1)[0]
   for batch in range(10):
      loss = a / (batch+1)
      wandb.log({"batch": batch + epoch * batch, "epoch": epoch, "loss": loss})
```

Here's what I see in the UI:
![Capture](https://user-images.githubusercontent.com/7469704/99898942-3d575f80-2cae-11eb-8711-70f1177d3e24.PNG)

And using the x-axis option int the UI I get this, for "batch":
![Capture](https://user-images.githubusercontent.com/7469704/99898960-59f39780-2cae-11eb-9579-721bad809cf6.PNG)

Or this, for "epoch":
![Capture](https://user-images.githubusercontent.com/7469704/99898972-72fc4880-2cae-11eb-8e88-6938a24b3738.PNG)

Where what I'm looking for is something like this:
![Capture](https://user-images.githubusercontent.com/7469704/99899045-de461a80-2cae-11eb-8880-d6ef49672d0f.PNG)

thanks for all the help

**Comment by cvphelps** [comment]
Hi Nivha, thanks for explaining your question so thoroughly. I'd recommend having a different metric name for each of your epochs, like 'loss-1', 'loss-2', etc. Then in the UI, click edit on a line plot — next to Y-axes, select multiple options to get them all to show up on the same chart.

Here's an example in [Colab](https://colab.research.google.com/drive/1FIVURdVCnpuc3u-e3LPa_T43_4-Rhah4?usp=sharing) with an example [page in the app](https://wandb.ai/carey/demo-multi-lines/runs/1elyyk75?workspace=user-carey)

```python
import wandb
import random
import math

wandb.init(project="demo-multi-lines")

for epoch in range(30):
  offset = random.random()
  for batch in range(100):
    acc = math.log(.1 + random.random() + batch) + offset
    wandb.log({f"acc-{epoch}": acc, "batch": batch})
```

![image](https://user-images.githubusercontent.com/6355078/100188508-3774a080-2e9f-11eb-869f-c27175d2acd9.png)

**Comment by ariG23498** [comment]
Hey @nivha 
We are closing this issue has a workaround. Please feel free to reopen the issue if the suggested workaround does not work for you. :smile:
